### PR TITLE
♻️ mass refactor: graphql + context library

### PR DESCRIPTION
**Pull request by ebullient**
- Move GraphQL GitHub data helpers/contexts into a separate library
- Submodule for notification/voting functions, still commonhaus-bot

### CODE REVIEW

File: bot-github-core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+
+    <!-- Using a parent for dependency versions,
+       code formatting, and plugins. -->
+    <parent>
+        <groupId>org.commonhaus.automation</groupId>
+        <artifactId>bot-parent</artifactId>
+        <version>${revision}</version>
+        <relativePath>../pom.xml</relativePath>
```
Comment (aalmiray) [inline bot-github-core/pom.xml:14]: This is the default value. Do you have to set it explicitly for a reason?

File: bot-github-core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    </parent>
+
+    <artifactId>bot-github-core</artifactId>
+
+    <name>Commonhaus Automation - GitHub Core</name>
+    <description>Query context + GraphQL assist</description>
+
+    <properties>
+        <!-- The following is set by mvnw in the parent (usually), but if you don't use mvnw... -->
+        <maven.multiModuleProjectDirectory>../</maven.multiModuleProjectDirectory>
```
Comment (aalmiray) [inline bot-github-core/pom.xml:24]: This property should be ser if there's a `.mvn` dir at the root of the reactor. Also, if the project has `mvnw` why not use it? (I know, I know) 😅

File: bot-github-core/src/main/java/org/commonhaus/automation/github/context/BotComment.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+public class BotComment {
+    final Pattern botCommentPattern;
+    final private String itemId;
+    private final String id;
+    private final int databaseId;
+    private final String url;
+    private String bodyString;
+
+    BotComment(Pattern botCommentPattern, String itemId, DataCommonComment comment) {
```
Comment (aalmiray) [inline bot-github-core/src/main/java/org/commonhaus/automation/github/context/BotComment.java:14]: Package private visibility in fields and constructor in public class with public methods. Yuck 😛

**Review by kenfinnigan — APPROVED** [review]

File: bot-github-core/pom.xml (hunk 1 above)
Comment (ebullient) [reply bot-github-core/pom.xml:14]: I seem to remember things being weird without it... at some point. ;)

File: bot-github-core/src/main/java/org/commonhaus/automation/github/context/BotComment.java (hunk 3 above)
Comment (ebullient) [reply bot-github-core/src/main/java/org/commonhaus/automation/github/context/BotComment.java:14]: I love package private fields. They are my favorite. ;)

Some things are best known w/in a package (not w/in an inheritance tree)

**Pull request merged**